parsing.go

```diff
@@ -131,50 +153,73 @@ func parseMarkdown(f string) ([]*Req, error) {
 					if level == lastHeadingLevel {
 						return nil, fmt.Errorf("requirement heading on line %d at same level as previous heading on line %d (%d): %q", lno, lastHeadingLine, level, line)
 					}
-					start = true
-				} else {
-					// No requirement ID on this heading. Will be ignored.
 				}
 			}
 
-			if end {
-				// Close the current requirement.
-				newReq, err := parseReq(reqBuf.String())
-				if err != nil {
-					return nil, err
-				}
-				reqs = append(reqs, newReq)
-				inReq = false
+			// If we're currently parsing a requirement and it's appropriate close it
+			if (inReq != None) && (headingHasReqID || level < reqLevel) {
+				reqs, err = parseMarkdownFragment(inReq, reqBuf.String(), reqs)
+				inReq = None
 			}
-			if start {
-				// Start a new requirement at the current line.
-				inReq = true
+
+			// If this is the start of a new requirement initialise data
+			if headingHasReqID {
+				inReq = Heading
 				reqLevel = level
 				reqLine = lno
 				reqBuf.Reset()
 				line = title
 			}
+			if level > 0 {
+				lastHeadingLevel = level
+				lastHeadingLine = lno
+			}
+		} else if reTableHeader.MatchString(line) {
+			// It's a requirements table
+			// If we're currently parsing a requirement close it
+			if inReq != None {
+				reqs, err = parseMarkdownFragment(inReq, reqBuf.String(), reqs)
```

Comment:
it's not serious, because it's still a small enough piece of code, but the duplication of the "close the existing requirement" makes me wary that in the future something will change. I think I would be tempted to keep a copy of the needed things for parseMarkdownFragment, and then move the close requirement code to just before "if inReq != None" so it applies to both types.

Reply:
I had a go at this but the result just ended up quite messy... I propose to wait and see if it becomes a problem =)